### PR TITLE
WIP: added wireguard input plugin

### PR DESCRIPTION
**Pull request by gitirabassi**
### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [] Has appropriate unit tests.

### CODE REVIEW

File: plugins/inputs/wireguard/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+```
+
+### Metrics
+
+- wireguard
+  - tags:
+    - name
+    - type
+    - serverpublikey
```
Comment (BondAnthony) [inline plugins/inputs/wireguard/README.md]: Spelling, `serverpublickey`

File: plugins/inputs/wireguard/README.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    - listenport
+    - firewallmark
+    - peerpublickey
+    - endpoint
+  - fiels:
```
Comment (BondAnthony) [inline plugins/inputs/wireguard/README.md]: `fields`

File: plugins/inputs/wireguard/README.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    - received_bytes
+    - transmit_bytes
+    - protocol
+    - last_hanshake_time
```
Comment (BondAnthony) [inline plugins/inputs/wireguard/README.md]: `last_handshake_time`

File: plugins/inputs/wireguard/README.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    - persisten_keepalive_interval
```
Comment (BondAnthony) [inline plugins/inputs/wireguard/README.md]: `last_handshake_time` and `persistent_keepalive_interval`

**Comment by danielnelson** [comment]
Related #6367

File: plugins/inputs/wireguard/README.md (hunk 1 above)
Comment (glinton) [reply plugins/inputs/wireguard/README.md]: We prefer snake cased tag/field names as well

**Comment by gitirabassi** [comment]
@danielnelson Before attempting any more work on this I want to make sure, since `golang.zx2c4.com/wireguard/wgctrl ` is the official wireguard client in Go now, that I will be ok and acceptable to import `github.com/mdlayher/genetlink` in the telegraf repository. This was the issue blocking #6367 . Personally I don't think neither wgctrl nor telegraf will be migrating netlink library. I believe it comes down to how bad to we want wireguard input plugin and I think it's quite important since it will be the next big vpn and we might just be a little bit behind given https://github.com/MindFlavor/prometheus_wireguard_exporter

**Comment by danielnelson** [comment]
Overall I agree with your assessment on the netlink library, though I'm perhaps less concerned about using the official client library and more optimistic that docker will do the switch for us someday. 

I'll look over these libraries soon, just my normal pass for really bad things, and get back with you.  Usually I don't find anything thats a problem.  Then let's work to update #6367 with any features you think need to be changed or added instead of continuing with this PR.

**Comment by gitirabassi** [comment]
I agree #6367 is better. Thanks and i’ll wait for your feedback

**Pull request closed without merging**